packages/flutter/lib/src/material/reorderable_list.dart

```diff
@@ -596,3 +596,27 @@ class _ReorderableListContentState extends State<_ReorderableListContent> with T
     });
   }
 }
+
+/// A global key that takes its identity from the object and uses a value of a
+/// particular type to identify itself.
+///
+/// The difference with GlobalObjectKey is that it uses [==] instead of [identical]
+/// of the objects used to generate widgets.
+@optionalTypeArgs
+class _GlobalObjectValueKey<T extends State<StatefulWidget>> extends GlobalObjectKey<T> {
+
+  const _GlobalObjectValueKey(this.value) : super(value);
+
+  final Object value;
+
+  @override
+  bool operator ==(Object other) {
+    if (other.runtimeType != runtimeType)
+      return false;
+    return other is _GlobalObjectValueKey<T>
+        && other.value == value;
+  }
+
+  @override
+  int get hashCode => super.hashCode;
+}
```

Comment:
The problem with this approach is it'll fail if there's _two_ `ReorderableListView`s with items that have the same local keys. I think the solution is that `_GlobalObjectValueKey` should have two values, one is the key we're proxying to, and the other is the _ReorderableListContentState to which they belong. We can probably rename it to `_ReorderableListChildGlobalKey` or something, with arguments `final Key subkey` and `final _ReorderableListContentState state` or something like that, to make it clearer what is going on.

Reply:
@Hixie 
That's a great idea! Thanks for the comment. I will apply it immediately.

Reply:
@Hixie Add `_ReorderableListContentState` as a value of the key can not solve the situation where multiple items use the same local key, which will lead to Multiple widgets used the same GlobalKey, such as:
```
  @override
  Widget build(BuildContext context) {
    return ReorderableListView(
      children: <Widget>[ InnerTest(key: ValueKey(0)), InnerTest(key: ValueKey(0)) ],
      onReorder: (_, __) {},
    );
  }
```
The reason why there is no problem here is its wrong implementation of `int get hashCode => identityHashCode(subKey);`

Reply:
Probably we could add the `index` into the values together.

Reply:
@xu-baolin Two children can never have the same key, so the current behavior is correct

Reply:
See: https://api.flutter.dev/flutter/foundation/Key-class.html

"Keys must be unique amongst the Elements with the same parent."

Reply:
I see, thank you.

Reply:
@PixelToast However, this change not an equivalent modification, and maybe cause ` Multiple widgets used the same GlobalKey` during layout phase. See https://github.com/flutter/flutter/pull/41338#issuecomment-538082979

Reply:
@xu-baolin No problem!

This PR is for fixing a specific bug and not extending functionality though, you are welcome to open a new issue to discuss any changes to functionality that would be nice to have.